src/txkube/test/test_authentication.py

```diff
@@ -171,13 +171,13 @@ def test_http_bearer_token_authorization(self):
         issues.  The header includes the bearer token from the service account
         file.  This works over HTTP.
         """
-        token = bytes(uuid4())
+        token = str(uuid4()).encode("ascii")
         request_bytes = self._authorized_request(token=token, headers=None)
 
         # Sure would be nice to have an HTTP parser.
         self.assertThat(
             request_bytes,
-            Contains(u"Authorization: Bearer {}".format(token).encode("ascii")),
+            Contains(b"Authorization: Bearer " + token),
```

Comment:
This is `bytes + unicode` on Python 3 as currently written, right?  That seems like a thing to avoid.

Is there a technical reason to avoid using `format` to construct text fragments like this one?  It seems like `format` is just the thing.

Reply:
`.format` for bytes was only introduced in PEP 461, and made it into Python 3.5.
I'm not sure how many versions of Python 3 you wanted to support in txkube.
`token` must be bytes, so doing `b"Authorization: Bearer " + token` seemed like the safest way to get things to work here consistently on Python 2.7 and Python 3.x